djangocms_navigation/models.py

```diff
@@ -27,20 +27,26 @@ class Meta:
 
 
 class MenuContent(models.Model):
-    title = models.CharField(verbose_name=_("title"), max_length=100)
     menu = models.ForeignKey(Menu, on_delete=models.PROTECT)
+    root = models.ForeignKey('djangocms_navigation.MenuItem')
```

Comment:
We should use a class on Foreign key ref rather than string and on_delete we should protect menucontent entry. Also, we should allow `null=true` as in most cases user will add menucontent entry first and then menuitem so user will come back to save root.

Reply:
I think we should create the MenuItem object automatically. @chaosk Would you rather this was done on the model level (i.e. on MenuContent save) or on the admin level? If on the model level I can add it to this PR, otherwise it'll be in mine or @webbyfox's next PR I guess.

Reply:
Think model's save is better here.

Reply:
@webbyfox So actually, defining foreign keys with strings is pretty standard: https://docs.djangoproject.com/en/2.1/ref/models/fields/#foreignkey It's also in some cases a good way to avoid circular imports and stuff like that. I can change it from a string to the class, but then I'll have to reorder the model classes and I actually thought they made more sense in the present order (as Menu is the "parent" of MenuContent, MenuContent of MenuItem).